qiskit/aqua/operators/list_ops/list_op.py

```diff
@@ -129,7 +129,10 @@ def primitive_strings(self) -> Set[str]:
 
     @property
     def num_qubits(self) -> int:
-        return self.oplist[0].num_qubits
+        num_qubits0 = self.oplist[0].num_qubits
+        if not all(num_qubits0 == op.num_qubits for op in self.oplist):
```

Comment:
```suggestion
        if not all(num_qubits0 == op.num_qubits for op in self.oplist[1:]):
```

Reply:
`self.oplist[1:]` is likely less efficient than `self.oplist` because the former allocates and makes a copy of the list, whereas the latter adds one dereference and integer comparison... On the other hand, efficiency is not a design criterion for aqua, and using the slice signals algorithmic intent.

Reply:
The fact that it ends up checking itself is fine - what we care about is that they are all the same, whichever one we pick as some reference.